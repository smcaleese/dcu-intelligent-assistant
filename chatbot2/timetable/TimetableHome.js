import React, { useState } from 'react';
import { StyleSheet, View, Text, TextInput } from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import { getClasses } from '../utils/GetClasses'
import CustomButton from '../components/CustomButton'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#f0f0f0',
        alignItems: 'center'
    },
    form: {
        backgroundColor: 'white',
        borderBottomLeftRadius: 10,
        borderBottomRightRadius: 10,
        padding: 20,
        width: '80%',
        justifyContent: 'center',
        alignItems: 'center'
    },
    titleContainer: {
        backgroundColor: "#4269E1",
        width: "80%",
        borderTopLeftRadius: 10,
        borderTopRightRadius: 10,
        padding: 5,
        marginTop: 50
    },
    titleText: {
        textAlign: 'center',
        fontSize: 24,
        color: "white"
    },
    courseCodeInput: {
        borderWidth: 1,
        paddingLeft: 10,
        width: '70%',
        margin: 20
    }
})

// source: https://github.com/react-native-datetimepicker/datetimepicker

export default function TimetableHome(props) {

    const [courseCode, setCourseCode] = useState('')
    const [date, setDate] = useState(new Date())
    const [show, setShow] = useState(false)

    //Handle change in selected date
    const handleChange = (event, newDate) => {
        console.log(event, newDate)
        const userDate = newDate || date
        setShow(false)
        setDate(userDate)
        console.log("userDate:", userDate)
    }

    //Fetch data from getClasses util
    const submit = () => {
        if(courseCode.length === 0) {
            return
        }
        getClasses(courseCode, date)
        .then((res) => {
            props.navigation.navigate("output", { 
                classesData: res.classesData, 
                dayOfTheWeek: res.dayOfTheWeek, 
                jsDate: res.jsDate,
                courseCode: courseCode
            })
        })
        .catch((err) => {console.log(err)})
    }

    return (
        <View style={styles.container}>
            <View style={styles.titleContainer}>
                <Text style={styles.titleText}>Timetable</Text>
            </View>
            <View style={styles.form}>
                <TextInput
                    style={styles.courseCodeInput}
                    placeholder={"Course code"}
                    value={courseCode}
                    onChangeText={newText => setCourseCode(newText)}
                />
                <CustomButton title="Choose Date" color="#4169e1" onPress={() => setShow(true)} />
                {show && <DateTimePicker
                    testID="dateTimePicker"
                    value={date}
                    mode="date"
                    display="default"
                    onChange={handleChange}
                />}
                <CustomButton title="Submit" color="green" onPress={submit} />
            </View>
        </View>
    )
}