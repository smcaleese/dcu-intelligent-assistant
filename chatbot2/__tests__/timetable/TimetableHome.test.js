import React from 'react';
import {create} from 'react-test-renderer';
import TimetableHome from '../../timetable/TimetableHome';

const screen = create(<TimetableHome />)

test('TimetableHome Snapshot Test', () => {
    expect(screen).toMatchSnapshot();
})