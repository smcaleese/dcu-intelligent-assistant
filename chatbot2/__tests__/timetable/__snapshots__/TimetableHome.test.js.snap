// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimetableHome Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f0f0f0",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4269E1",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "marginTop": 50,
        "padding": 5,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      Timetable
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "justifyContent": "center",
        "padding": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Course code"
      rejectResponderTermination={true}
      style={
        Object {
          "borderWidth": 1,
          "margin": 20,
          "paddingLeft": 10,
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4169e1",
          "borderRadius": 5,
          "margin": 10,
          "opacity": 1,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Choose Date
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "borderRadius": 5,
          "margin": 10,
          "opacity": 1,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
